codalab/lib/worksheet_util.py

```diff
@@ -80,6 +83,126 @@ def worksheet_line(description, uuid):
     return '[%s]{{%s}}' % (description, uuid)
 
 
+
+############################################################
+
+
+class WorksheetResolver(object):
```

Comment:
@percyliang by merging with canonicalize.py, you mean we don't actually want to "create a new Python file"?